### ISSUE TITLE
Can we get feedback when compiling, about what FSC version is being used?

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#13816]
This may be more relevant to people working with developing the compiler itself, but I often find myself wondering what compiler is being configured from my Visual Studio. Having both VS 2022, older version and VS 2022 Preview, plus the one compiled by the F# repo, I'm often not 100% sure what version I'm using.

This is even more relevant when reporting bugs. People tend to be behind on updates, so having the FSC (or VS or whatever) dump the version of FSC just before compiling.

Another use case is in CI, which can have slightly different versions of FSC. As such, dumping the version for comparison may help analyse spurious bugs.

## Proposed solution

Instead of this:

```
Build started...
1>------ Build started: Project: My.Extensions.Actions, Configuration: Debug Any CPU ------
2>------ Build started: Project: My.Extensions.Security, Configuration: Debug Any CPU ------
```

I'd like to see this instead:

```
Build started using fsc version 6.0.12345-beta45678...
1>------ Build started: Project: My.Extensions.Actions, Configuration: Debug Any CPU ------
2>------ Build started: Project: My.Extensions.Security, Configuration: Debug Any CPU ------
```

It may also be reported on each individual project, if that makes more sense.

## Describe alternatives you've considered

Well, I tried using `--version` on `fsc` but that crashed Visual Studio (https://github.com/dotnet/fsharp/issues/13549). This has meanwhile been fixed (https://github.com/dotnet/fsharp/pull/13702), but the solution suppresses the version output, as opposed to showing it.

**Comment by vzarytovskii** [comment]
Yeah, I guess it makes sense to add it.

**Comment by psfinaki** [comment]
Hmmm good idea, thanks for specifying the actual usecases for this. 
@vzarytovskii assigning this on me for now - but let's plan this separately later.

**Comment by abelbraaksma** [comment]
Thanks, @psfinaki. If you need help with this, like testing something or whatnot, let me know!

**Comment by dsyme** [comment]
I don't think this is possible, as `Build started...` is emitted by MSBuild and is not F# specific.  

You can turn on detailed MSBuild output in the VS options, I think that's your best bet.

**Comment by vzarytovskii** [comment]
> I don't think this is possible, as `Build started...` is emitted by MSBuild and is not F# specific.  
> 
> You can turn on detailed MSBuild output in the VS options, I think that's your best bet.  
> 
> I don't know if there's any way for us to increase the output shown in the Build window by default.  Plus the version of F# tooling being used is available for the Help --> About menu?
> 
> 

I guess we can add version output to the FCS msbuild task, which will just read it from the assembly?

**Comment by abelbraaksma** [comment]
> Plus the version of F# tooling being used is available for the Help --> About menu?

For a while now, this has always shown a beta version of F#, even though you are in a normal (non pre-release) VS 2022. I've reported it a few years back, but it is still there. That version has no semblance whatsoever with the FSC version.

> Visual F# Tools   17.1.0-beta.22363.4+1b94f89d4d1f41f20f9be73c76f4b229d4e49078

Also, the concrete FSC executable used is _very hard_ to figure out. So if there's some way of getting this info, it would be so, so helpful. Since output of FSC is shown in the output window, I'd guess just dumping the current version as part of it shouldn't be too hard.
